modules/gateway/peers.go

```diff
@@ -442,14 +442,9 @@ func (g *Gateway) managedConnectNewPeer(conn net.Conn, remoteVersion string, rem
 	return nil
 }
 
-// Connect establishes a persistent connection to a peer, and adds it to the
-// Gateway's peer list.
-func (g *Gateway) Connect(addr modules.NetAddress) error {
-	if err := g.threads.Add(); err != nil {
-		return err
-	}
-	defer g.threads.Done()
-
+// managedConnect establishes a persistent connection to a peer, and adds it to
+// the Gateway's peer list.
+func (g *Gateway) managedConnect(addr modules.NetAddress) error {
```

Comment:
Why separate `Connect` into `Connect` and `managedConnect`? The only other place `managedConnect` is used outside of `Connect` it calls `tg.Add(); defer tg.Done()` just like `Connect` does.